translation/normalization.py

```
@@ -0,0 +1,270 @@
+
+        1. Consistent whitespace.
+        2. Basic alpha-equivalence for bound variables in quantifiers.
+        3. Parenthesizes expressions to avoid ambiguity.
+    """
+
+    def __init__(self) -> None:
+        """Create a new CBMCNormalizer."""
+        self._scopes: list[dict[str, str]] = []
+        self._counter = 0
+
+    def normalize(self, node: Any) -> str:
```

Ruff (0.14.13)

Dynamically typed expressions (typing.Any) are disallowed in `node`

(ANN401)